x-pack/legacy/plugins/ml/server/routes/data_frame_analytics.js

```diff
@@ -4,171 +4,298 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { callWithRequestFactory } from '../client/call_with_request_factory';
-import { wrapError } from '../client/errors';
+import { schema } from '@kbn/config-schema';
 import { analyticsAuditMessagesProvider } from '../models/data_frame_analytics/analytics_audit_messages';
+import { licensePreRoutingFactory } from '../new_platform/licence_check_pre_routing_factory';
+import { dataAnalyticsJobConfigSchema } from '../new_platform/data_analytics_schema';
 
-export function dataFrameAnalyticsRoutes({ commonRouteConfig, elasticsearchPlugin, route }) {
-  route({
-    method: 'GET',
-    path: '/api/ml/data_frame/analytics',
-    handler(request) {
-      const callWithRequest = callWithRequestFactory(elasticsearchPlugin, request);
-      return callWithRequest('ml.getDataFrameAnalytics').catch(resp => wrapError(resp));
-    },
-    config: {
-      ...commonRouteConfig,
+export function dataFrameAnalyticsRoutes({ xpackMainPlugin, router }) {
+  async function dfAnalyticsGetAllHandler(context, request, response) {
+    try {
+      const results = await context.ml.mlClient.callAsCurrentUser('ml.getDataFrameAnalytics');
+      return response.ok({
+        body: { ...results },
+      });
+    } catch (e) {
+      // Case: default
+      return response.internalError({ body: e });
```

Comment:
This will expose the stack trace of `e` which could reveal information about the server like the location of files. This should rather be `response.internalError({body: e.message});` (also in other places in the PR)

Reply:
Thanks for taking a look! 😄  
Updated here: 66d9d97048c385885a158804899124f469390672